certbot/certbot/_internal/auth_handler.py

```diff
@@ -79,7 +96,9 @@ def handle_authorizations(self, orderr, config, best_effort=False, max_retries=3
                 logger.info('Attempting to clean up outstanding challenges...')
                 raise error
             # All challenges should have been processed by the authenticator.
-            assert len(resps) == len(achalls), 'Some challenges have not been performed.'
+            # TODO: Remove the cast once certbot package is fully typed
+            assert len(cast(Sized, resps)) == len(achalls), \
+                'Some challenges have not been performed.'
```

Comment:
Well, it seems that the typing of `Authenticator.perform` deviated from my global plan at some point, because originally I was claiming that it returns `List[ChallengeResponse]`, not `Iterable[ChallengeResponse]`. Let's fix that.

Reply:
No in fact the incoming part1 and part2 fixed that in fact. We can just remove the `cast` now.